mycore-orcid2/src/main/java/org/mycore/orcid2/oauth/MCRORCIDOAuthClient.java

```diff
@@ -109,7 +112,21 @@ private void handleError(Response response) {
         throw new MCRORCIDRequestException(response);
     }
 
+    public record Settings(
+        String baseUrl,
+        String clientId,
+        String clientSecret) {
+
+        public Settings validate() {
```

Comment:
I suggest to validate the params in a custom custructor:
https://www.baeldung.com/java-records-custom-constructor#how-to-make-a-custom-constructor-for-a-java-record